### PR TITLE
Kf/critical gradient

### PR DESCRIPTION
**Pull request by kfrybes**
Branch for developing a critical gradient optimization metric to reduce turbulent transport. Theory for critical gradient is taken from https://journals.aps.org/prresearch/pdf/10.1103/PhysRevResearch.4.L032028 .

### CODE REVIEW

File: desc/objectives/_stability.py
Comment (kfrybes) [inline desc/objectives/_stability.py:1]: that's a mistake

File: critical_gradient_jax.ipynb
Comment (kfrybes) [inline critical_gradient_jax.ipynb:1]: Notebook for testing jax implementation of critical gradient

File: Critical_gradient_optimization.ipynb
Comment (kfrybes) [inline Critical_gradient_optimization.ipynb:1]: Notebook for testing critical gradient optimization

File: zonal_flow.ipynb
Comment (kfrybes) [inline zonal_flow.ipynb:1]: Notebook for fitting zonal flows, not up to date

File: Critical_gradient2.ipynb
Comment (kfrybes) [inline Critical_gradient2.ipynb:1]: Notebook for testing critical gradient fitting, not up to date

**Comment by ddudt** [comment]
@kfrybes see [our contributing guidelines](https://github.com/PlasmaControl/DESC/blob/master/CONTRIBUTING.rst#styleguides) for info on installing `black` and `pre-commit`.

**Comment by dpanici** [comment]
Just a note, please check the [styleguide](https://github.com/PlasmaControl/DESC/blob/master/CONTRIBUTING.rst#python-styleguide) part of our contribution guidelines and install pre-commit so you can catch these formatting and other warnings! that way you can avoid all of these comments on the PR

**Comment by kfrybes** [comment]
yep, sorry, I made a bit of a mess

**Comment by dpanici** [comment]
> yep, sorry, I made a bit of a mess

I've been there (#579 has over 500 comments)...

File: desc/objectives/_turbulence.py [hunk 1]
```diff
@@ -0,0 +1,676 @@
...
+            M=eq.M_grid,
+            N=eq.N_grid,
+            NFP=eq.NFP,
+        )
+        self._iota_keys = ["iota", "iota_r", "shear"]
+        iota_profiles = get_profiles(self._iota_keys, obj=eq, grid=iota_grid)
+        iota_transforms = get_transforms(self._iota_keys, obj=eq, grid=iota_grid)
+
+        # Separate grid to calculate the right length scale for normalization
+        len_grid = LinearGrid(rho=1.0, M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP)
```
Comment (unalmis) [inline desc/objectives/_turbulence.py:137]: > Understand why effective radius returned by optimizer is half of the vlaue returned by the compute function

Probably need quadrature grid. See #1424
